File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        Arrays.sort(nums);
        int count=0;
        for(int i=0; i<nums.length-1; i++){
            if(nums[i]==nums[i+1])
                count++;
            if(count >= nums.length/2)  return nums[i];     
        }
        return nums[0];
    }
}

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        # Creating a dictionary to store count of elements
        dic = {}
        #iterating through list
        for ch in nums:
            #each unique element is stored in the dic, ele as key and freq as value
            dic[ch] = dic.get(ch, 0) + 1

        # Find element with maximum frequency
        for key, val in dic.items():
            if val > len(nums) // 2:
                return key